purchase_origin_link_sale/models/purchase.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _selection_reference_model(self):
+        res = super()._selection_reference_model() + [("sale.order", "Sale Order")]
+        return res
+
+    @api.depends(lambda x: x._get_depends_compute_source_doc_ref())
+    def _compute_source_doc_ref(self):
+        super()._compute_source_doc_ref()
+        for rec in self:
+            so = self.env["sale.order"].search([("name", "like", rec.origin)], limit=1)
+            if so:
+                rec.origin_reference = "{},{}".format(so._name, so.id or 0)
```

Comment:
```suggestion
            so = self.env["sale.order"].search([("name", "like", rec.origin)])
            if len(so) == 1:
                rec.origin_reference = "{},{}".format(so._name, so.id or 0)
```

Reply:
Maybe I would put it like this so that, if there are multiple origin SO, we don't loose all the references, and we just make it clickable if the source is only a single SO

Reply:
Sure, maybe could be better this way instead the other one.